csrc/ir/nodes.cpp

```diff
@@ -4378,15 +4378,14 @@ std::vector<PolymorphicValue> SdpaFwdOp::evaluate(
     output = output.slice(-1, 0, last_dim_size);
   }
 
-  // Query and key seq len are of type c10::SymInt -> convert them to int for
-  // Polymorphic Value
+  // Query and key seq len are of type c10::SymInt -> convert them to CPU scalar tensors to support adding them as fusion outputs.
   return {
       output,
       log_sumexp,
       cum_seq_q,
       cum_seq_k,
-      *query_seq_len.maybe_as_int(),
-      *key_seq_len.maybe_as_int(),
+      at::scalar_tensor(*query_seq_len.maybe_as_int(), at::dtype(at::kLong)),
```

Comment:
`at::tensor` creates a 1D tensor, so using `scalar_tensor`.

Reply:
Please keep the code as-is, scalar_tensor is easier to read.

Meanwhile, `at::tensor(*query_seq_len.maybe_as_int(), at::dtype(at::kLong))` should give you the same scalar tensor.

Reply:
I was getting a 1D tensor instead of 0-dim, will check again.
I agree with you that `scalar_tensor` is more clear.